plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -194,6 +221,60 @@ public IcebergTableHandle getTableHandle(ConnectorSession session, SchemaTableNa
                 Optional.ofNullable(nameMappingJson));
     }
 
+    @Override
+    public boolean isSupportedVersionType(ConnectorSession session, SchemaTableName tableName, PointerType pointerType, io.trino.spi.type.Type versioning)
+    {
+        switch (pointerType) {
+            case TEMPORAL:
+                return versioning instanceof TimestampWithTimeZoneType || versioning instanceof TimestampType;
+            case TARGET_ID:
+                return versioning == BIGINT;
+        }
+        throw new TrinoException(NOT_SUPPORTED, "Version type " + versioning + " is not supported");
+    }
+
+    private Optional<Long> getSnapshotIdFromVersion(Table table, Optional<ConnectorTableVersion> versionOptional)
+    {
+        Optional<Long> snapshotId = Optional.empty();
+        if (versionOptional.isPresent()) {
+            ConnectorTableVersion version = versionOptional.get();
+
+            switch (version.getPointerType()) {
+                case TEMPORAL:
+                    long epochMillis;
+                    io.trino.spi.type.Type versionType = version.getVersionType();
+                    if (versionType instanceof TimestampWithTimeZoneType) {
+                        if (((TimestampWithTimeZoneType) versionType).isShort()) {
+                            epochMillis = getEpochMillisFromShortTimestampWithTimeZone((long) version.getVersion());
```

Comment:
why not just `unpackMillisUtc(version.getVersion())`

Reply:
I think that ignores the `timeZoneKey` encoded in the long

Reply:
That is good right? Don't you want to get number of millis since epoch (in UTC)?

Reply:
updated